File: java/quantile-estimator/src/main/java/org/mydotey/quantile/classic/ClassicQuantileEstimatorConfig.java
package org.mydotey.quantile.classic;

import java.util.Comparator;
import java.util.Objects;

import org.mydotey.quantile.value.Calculator;

/**
 * @author koqizhao
 *
 * Apr 1, 2018
 */
public class ClassicQuantileEstimatorConfig<T> {

    private Comparator<T> _comparator;
    private Calculator<T> _calculator;

    public ClassicQuantileEstimatorConfig(Comparator<T> comparator, Calculator<T> calculator) {
        Objects.requireNonNull(comparator, "comparator is null");
        Objects.requireNonNull(calculator, "calculator is null");

        _comparator = comparator;
        _calculator = calculator;
    }

    public Comparator<T> getComparator() {
        return _comparator;
    }

    public Calculator<T> getCalculator() {
        return _calculator;
    }

}

File: dotnet/src/MyDotey.QuantileEstimator/Classic/ClassicQuantileEstimatorConfig.cs
using System;
using System.Collections.Generic;

using MyDotey.Quantile.Value;

/**
 * @author koqizhao
 *
 * Apr 1, 2018
 */
namespace MyDotey.Quantile.Classic
{
    public class ClassicQuantileEstimatorConfig<T>
    {
        public virtual IComparer<T> Comparer { get; }
        public virtual ICalculator<T> Calculator { get; }

        public ClassicQuantileEstimatorConfig(IComparer<T> comparer, ICalculator<T> calculator)
        {
            if (comparer == null)
                throw new ArgumentNullException("comparer is null");

            if (calculator == null)
                throw new ArgumentNullException("calculator is null");

            Comparer = comparer;
            Calculator = calculator;
        }
    }
}